java/java.lsp.server/vscode/src/panels/SshGuidePanel.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as vscode from "vscode";
+import { GuidePanel } from "./GuidePanel";
+import { dummyKeyPathLocation, sshConfigLocation, toggleGuideFor } from "./guidesUtil";
+
+export type SshGuidePanelProps = {
+    publicIp: string;
+    ocid: string;
+}
+
+export const viewType: string = "sshGuide";
+const templatePath: string = "ssh-guide.handlebars";
+
+export class SshGuidePanel extends GuidePanel {
+
+    public static currentPanel: SshGuidePanel | undefined;
+
+    public static createOrShow(context: vscode.ExtensionContext, props: SshGuidePanelProps) {
+        if (SshGuidePanel.currentPanel) {
+            SshGuidePanel.currentPanel.panel.reveal();
+            return;
+        }
+
+        SshGuidePanel.currentPanel = new SshGuidePanel(context, props);
+    }
+
+    constructor(context: vscode.ExtensionContext, props: SshGuidePanelProps) {
+        const { ocid, publicIp } = props;
+        super(context, viewType, templatePath, {
+            ocid,
+            publicIp,
+            dummyKeyPathLocation,
+            sshConfigLocation,
+            showGuide: "",
+        });
+    }
+
+    async messageHandler(message: any): Promise<void> {
```

Comment:
Just a suggestion: given that superclass has `getSource` that assumes current guide's style (subclasses can override),  the same principle may be used with `messageHandler` stock implementation: can be moved to the base class; if other javascript than `guide.js` is returned from `getSources()` it can use a differet msg command and an appropriate subclass can handle it.

Reply:
I was thinking about it, but then parent `GuidePanel.ts` becomes aware of specific `properties` that are passed to it